torch/csrc/dynamo/compiled_autograd.h

```diff
@@ -900,6 +900,499 @@ class SwapSavedVariables {
   StashedVars<at::IValue> stashed_ivalues;
 };
 
+// NOTE: [Compiled Autograd and backward functions]
+// Built-in autograd nodes have functional apply variants
+// (e.g. MulBackward0_apply_functional). Compiled Autograd's initial graph
+// capture wants to take a variant of this function and proxy it into the graph.
+// Every autograd node defines an apply_with_saved function, that when invoked,
+// proxys a call to a function into the Compiled Autograd graph.
+//
+// Some requirements that we have are:
+// - The proxy'ed function must have inputs that are FX-graphable types.
+// - Windows has a DLL symbol limit of 65536.
+// - Node::apply_with_saved is in libtorch_cpu which does not have direct access
+// to Python
+//
+// There were multiple ways to skin the cat, but what we end up doing is:
+// - for e.g. MulBackward0_apply_functional, we create a new C++ function
+// MulBackward0_apply_functional_ivalue that accepts vector<IValue>.
```

Comment:
minor perf warning: it is known to be pretty slow to convert between `vector<IValue>` and `vector<Tensor>`, so in any cases where we need to run this code at runtime (e.g. due to a graph break), we can expect this to be slower if the user's custom backward op accepts any `vector<Tensor>` arguments. Mostly a PSA - the usually "fix" is to change the code that accepts `vector<Tensor>` to instead accept `ITensorListRef` (which doesn't require an extra conversion / reallocating a bunch of `at::Tensor` objects)